### DISCUSSION TITLE
Does tailwind support for grouping CSS properties by type?

### DISCUSSION THREAD

**Discussion by duyphanz** [tailwindlabs/tailwindcss#5737] [Help] [answered]
I often order my CSS properties by grouping them by type, an example here: https://css-tricks.com/poll-results-how-do-you-order-your-css-properties/

So, I'd to know if tailwind supports this in some way? A long class name chain seems to somehow decrease the developer experience. 

E.g

```html
<div
    twClass="px-10 mx-10"
    twClass="bg-red-100 text-green-100"
    twClass:hover="bg-gray-100"
/>
```

**Comment by simonswiss** [comment, marked as answer]
This is something we've discussed a few times, but there are many different opinions in the community on how the sorting should occur.

I personally started enjoying how Prettier (in recent versions) groups "similar" properties in on the same line (variants, for example), after initially _hating_ it. You'd think I would have learned that it's normal to reject change at first sight 😅

Another potential solution, if you're on vscode, is to use the Headwind extension, which will make some decisions for you and enforce consistency in the class order:

https://marketplace.visualstudio.com/items?itemName=heybourn.headwind

Hope it helps 👍

**Reply by duyphanz** [reply]
Thanks for your help! I'm new to Atomic CSS and Tailwind too. It'd be great if the official Tailwind site has a section mentioning or linking to these related discussions someday. I will help so much for people like me. Thank you once again.
